### PR TITLE
feat: support splat route config file

### PR DESCRIPTION
**Pull request by yimingjfe**
### <samp>🤖 Generated by Copilot at 63977ee</samp>

This pull request adds a new feature to the `@modern-js/app-tools` package that allows users to create a splat route config file (`$.config`) for any route path and specify some route options. It also adds the Chinese translation for this feature.

### <samp>🤖 Generated by Copilot at 63977ee</samp>

*  Update changeset file to indicate patch version update and new features for `@modern-js/app-tools` package ([link](https://github.com/web-infra-dev/modern.js/pull/4673/files?diff=unified&w=0#diff-3b029eca74ddfab4136f12538bf85583986760c5d0077f72cfb6c9d4c4db0e59R1-R6))
*  Add `SPLATE_CONFIG_FILE` constant to `NESTED_ROUTE` object in `constants.ts` file to represent splat route config file name ([link](https://github.com/web-infra-dev/modern.js/pull/4673/files?diff=unified&w=0#diff-b02a3b2a5a6c983ffc4a0be8f2783d9fd7eb47f3d82d9f66edf48494d56e7085R50))
*  Modify `walk` function in `nestedRoutes.ts` file to support reading and applying splat route config file for nested routes ([link](https://github.com/web-infra-dev/modern.js/pull/4673/files?diff=unified&w=0#diff-89756710513f64f72aaf997fdd2a0179fd26970e9d18e579b4648498933ce1cdR160), [link](https://github.com/web-infra-dev/modern.js/pull/4673/files?diff=unified&w=0#diff-89756710513f64f72aaf997fdd2a0179fd26970e9d18e579b4648498933ce1cdR261-R266), [link](https://github.com/web-infra-dev/modern.js/pull/4673/files?diff=unified&w=0#diff-89756710513f64f72aaf997fdd2a0179fd26970e9d18e579b4648498933ce1cdR292-R294))
  * Declare `splatConfigFile` variable to store splat route config file path if it exists in current directory ([link](https://github.com/web-infra-dev/modern.js/pull/4673/files?diff=unified&w=0#diff-89756710513f64f72aaf997fdd2a0179fd26970e9d18e579b4648498933ce1cdR160))
  * Assign splat route config file path to `splatConfigFile` variable if current item is splat route config file and current route has no config property ([link](https://github.com/web-infra-dev/modern.js/pull/4673/files?diff=unified&w=0#diff-89756710513f64f72aaf997fdd2a0179fd26970e9d18e579b4648498933ce1cdR261-R266))
  * Assign `splatConfigFile` variable to `config` property of `splatRoute` object if it has a value ([link](https://github.com/web-infra-dev/modern.js/pull/4673/files?diff=unified&w=0#diff-89756710513f64f72aaf997fdd2a0179fd26970e9d18e579b4648498933ce1cdR292-R294))

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [x] I have updated the documentation.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: .changeset/quick-seas-sin.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@modern-js/app-tools': patch
+---
+
+feat: support splat route config file
+feat: 支持通配路由配置文件
```
Comment (chenjiahan) [inline .changeset/quick-seas-sin.md:6]: should we document this feature?

**Review by clChenLiang — APPROVED** [review]

File: .changeset/quick-seas-sin.md (hunk 1 above)
Comment (yimingjfe) [reply .changeset/quick-seas-sin.md:6]: done~

**Review by clChenLiang — APPROVED** [review]

File: packages/document/main-doc/docs/en/guides/basic-features/data-fetch.mdx [hunk 2]
```diff
@@ -13,7 +14,7 @@ It should be noted that these APIs do not help applications initiate requests, b
 Modern.js recommends using [conventional routing](/guides/basic-features/routes) for routing management. Through Modern.js's [conventional (nested) routing](/guides/basic-features/routes#conventional-routing), each routing component (`layout.ts` or `page.ts`) can have a same-named `loader` file. The `loader` file needs to export a function that will be executed before the component is rendered to provide data for the routing component.
 
 :::info
-Modern.js v1 supports fetching data via [useLoader](/guides/basic-features/data-fetch.html#useloader-(old-version)), which is no longer the recommended usage. We do not recommend mixing the two except during the migration process.
+Modern.js v1 supports fetching data via [useLoader](</guides/basic-features/data-fetch.html#useloader-(old-version)>), which is no longer the recommended usage. We do not recommend mixing the two except during the migration process.
```
Comment (chenjiahan) [inline packages/document/main-doc/docs/en/guides/basic-features/data-fetch.mdx:17]: The useLoader link is broken

Comment (yimingjfe) [reply packages/document/main-doc/docs/en/guides/basic-features/data-fetch.mdx:17]: I have verified this link, and it can jump normally

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**